### DISCUSSION TITLE
Clear recent projects

### DISCUSSION THREAD

**Discussion by Southclaws** [zed-industries/zed#6703] [Feature Requests]
I accidentally opened my drive's root as a "project" and now every time I open Zed it opens `/` and tries to index my entire computer!

Is there a way to clear the recent projects list? I can't find anything in the zed/ config folder. Is it stored in some system wide config registry on mac/windows?

**Comment by NicoAdrian** [comment]
Have the same problem. Even when clicking on "Remove from project", opening Zed again still opens it

**Comment by SomeoneToIgnore** [comment]
There's no user-facing way for that yet.

Whatever survives between Zed restarts, is stored in the `~/Library/Application Support/Zed/db/${INSTANCE_NAME}/db.sqlite` file.
Specifically, this is stored in `workspaces`, paths are in `workspace_location`.

Those are blobs, though but some GUI might help with browsing it (notice the right side):
![image](https://github.com/zed-industries/zed/assets/2690773/fb14dfae-2256-4628-be6b-caa9cb436b01)

so you can alter entries there (or remove the whole db file), or you can submit a PR to fix those.
We get the values from db using this: https://github.com/zed-industries/zed/blob/8263da02bd65e50611a4a70199273f49929fd8ea/crates/workspace/src/persistence.rs#L423-L430 and there's already a `delete_stale_workspace` method for removing a workspace by ID, so all that's needed — some gui adjustments in the `recent_projects.rs` that call this method.

**Reply by Southclaws** [reply]
Awesome thanks! I'll take a crack at doing a PR to add this as a command palette item if that's cool!

**Reply by NicoAdrian** [reply]
@Southclaws Great ! Would be cool if you could tell me here if/when you did it :-) 
Thanks

**Reply by Southclaws** [reply]
Unfortunately I didn't manage to get it figured out in time, bit too busy annoyingly!

**Reply by ikurotime** [reply]
Hello! If its possible, I will try to submit this, I already managed to locate the proper files. Any guidance / tip is welcomed, though!

**Reply by SomeoneToIgnore** [reply]
To note, there's some discussion happening here: https://github.com/zed-industries/zed/issues/7426
